spyder/plugins/ipythonconsole/comms/kernelcomm.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+In addition to the remote_call mechanism implemented in CommBase:
+ - Send a message to a debugging kernel
+"""
+import logging
+
+from qtpy.QtCore import QEventLoop, QObject, QTimer, Signal
+
+from spyder_kernels.comms.commbase import CommBase
+from spyder_kernels.py3compat import TimeoutError
```

Comment:
Please move this definition of TimeoutError to the `py3compat` module in Spyder. We don't mix py3compat modules of our packages.